core/drivers/crypto/hisilicon/hpre_ecc.c

```diff
@@ -0,0 +1,967 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, HiSilicon Technologies Co., Ltd.
+ * Kunpeng hardware accelerator hpre ecc algorithm implementation.
+ */
+#include <drvcrypt.h>
+#include <drvcrypt_acipher.h>
+#include <hpre_main.h>
+#include <hpre_ecc.h>
+#include <initcall.h>
+#include <malloc.h>
+#include <rng_support.h>
+#include <stdlib_ext.h>
+#include <string.h>
+#include <string_ext.h>
+#include <trace.h>
+
+struct hpre_ecc_curve {
+	uint32_t id;
+	uint32_t key_bits;
+	uint8_t *p;
+	uint8_t *a;
+	uint8_t *b;
+	uint8_t *x;
+	uint8_t *y;
+	uint8_t *n;
+};
+
+/* NID_X9_62_prime192v1 */
+static uint8_t g_prime192v1_p[] = {
```

Comment:
`static const`?

Reply:
![image](https://github.com/OP-TEE/optee_os/assets/118411745/88fc1098-68fc-4fc0-b05f-81b2a4b8689e)
cannot add const

Reply:
Is it possible to change `struct hpre_ecc_curve` to use const pointers instead or is that struct sometimes used for output too?

It would be advantageous to make this data const since it will be protected from accidental overwrite.